File: Level 0/코딩 기초 트레이닝/정수를 나선형으로 배치하기/solution.py
def solution(n):
    move = [[1, 0], [0, 1], [-1, 0], [0, -1]]
    x, y, m = 0, 0, 0

    answer = [[0 for _ in range(n)] for _ in range(n)]

    for i in range(1, n**2 + 1):
        answer[y][x] = i
        if (
            x + move[m][0] >= n
            or y + move[m][1] >= n
            or answer[y + move[m][1]][x + move[m][0]]
        ):
            m = (m + 1) % len(move)
        x, y = x + move[m][0], y + move[m][1]

    return answer

File: Level 0/코딩 기초 트레이닝/정수를 나선형으로 배치하기/solution.js
function solution(n) {
  const move = [
    [1, 0],
    [0, 1],
    [-1, 0],
    [0, -1],
  ];
  let x = 0;
  let y = 0;
  let m = 0;

  const answer = Array.from({ length: n }, () => Array(n).fill(0));

  for (let i = 1; i <= n ** 2; i++) {
    answer[y][x] = i;
    if (
      x + move[m][0] >= n ||
      y + move[m][1] >= n ||
      answer[y + move[m][1]][x + move[m][0]] !== 0
    ) {
      m = (m + 1) % move.length;
    }
    x = x + move[m][0];
    y = y + move[m][1];
  }

  return answer;
}
